sim/sim.go

```diff
@@ -152,12 +139,15 @@ func (s *Simulation) initParticipants() error {
 	// Limit adversaryCount to exactly one for now to reduce LOC up for review.
 	// Future PRs will expand this to support a group of adversaries.
```

Comment:
Nit: I think you can just remove this, and the TODO above and `adversaryCount`. Just note there's at most one adversary (though it can have arbitrary power).